jax/_src/lax/parallel.py

```diff
@@ -420,14 +400,15 @@ def _psum_translation_rule(c, *args, axis_name, axis_index_groups, axis_env,
   for dtype, (indices, dtype_args) in sorted(args_by_type.items()):
     is_complex = dtypes.issubdtype(dtype, np.complexfloating)
     n = len(dtype_args)
-    if is_complex:
+    if is_complex and prim is lax.add_p:
+      # we handle complex-dtype sum-reduction directly as a special case
```

Comment:
It would be good to add some extra rationale for why is it done this way (if you remember).

Reply:
Good point. I think the reason is/was that XLA did/didn't support collectives on complex values, but if that's the case then I should link a bug.

Reply:
Aha, it's b/141575627, where some folks commented: "The reason that the GPU doesn't support complex(AllReduce) is because nccl library doesn't support complex data types". Still, XLA:GPU could solve this for us.

I'll link the bug in a comment. Nice catch!